stream-chat-swift-docs/docs/guides/working-with-members.md

```diff
@@ -1,3 +1,223 @@
+
+### Filter by user.name exact match:
+```swift
+let memberListController = chatClient.memberListController( 
+    query: .init(
+        cid: .init(type: .messaging, id: "general"),
+        filter: .equal(.name, to: "Lando")
+    )
+) 
+```
+
+### Filter by part of the user.name
+
+```swift
+// Returns Lando, Conan etc.
+let memberListController = chatClient.memberListController( 
+    query: .init(
+        cid: .init(type: .messaging, id: "general"),
+        filter: .query(.name, text: "an")
+    ) 
+) 
+```
+
+### Autocomplete users by user.name
+
+```swift
+// TODO explanation?
+let memberListController = chatClient.memberListController( 
+    query: .init(
+        cid: .init(type: .messaging, id: "general"),
+        filter: .autocomplete(.name, text: "La")
+    ) 
+) 
+```
+
+### Query a member by id
+
+```swift
+let memberListController = chatClient.memberListController( 
+    query: .init(
+        cid: .init(type: .messaging, id: "general"),
+        filter: .equal(.id, to: "user-id")
+    )
+) 
+```
+
+### Query several members by id
+
+```swift
+let memberListController = chatClient.memberListController( 
+    query: .init(
+        cid: .init(type: .messaging, id: "general"),
+        filter: .in(.id, values: ["first-user", "second-user"])
+    ) 
+) 
+```
+
+### Query channel moderators 
+
+```swift
+let memberListController = chatClient.memberListController( 
+    query: .init(
+        cid: .init(type: .messaging, id: "general"),
+        filter: .equal(.isModerator, to: true)
+    )
+) 
+```
+
+### Query banned members
+
+```swift
+let memberListController = chatClient.memberListController( 
+    query: .init(
+        cid: .init(type: .messaging, id: "general"),
+        filter: .equal(.isBanned, to: true)
+    ) 
+) 
+```
+
+### Query members with pending invites 
+
+```swift
+let memberListController = chatClient.memberListController( 
+    query: .init(
+        cid: .init(type: .messaging, id: "general"),
+        filter: .equal("invite", to: "pending")
+    ) 
+) 
+```
+
+## Pagination and ordering
+
+Since there potentially can be large amount of members in a channel, querying members supports pagination.
+By default members are ordered from oldest to newest and can be paginated using offset-based pagination:
+
+```swift
+memberListController.loadNextMembers(limit: 10) { error in 
+    // handle error / access members 
+    print(error ?? memberListController.members) 
+}
+```
+
+## Adding/removing members
+
+```swift
+let controller = chatClient.channelController(for: .init(type: .messaging, id: "general")) 
+controller.addMembers(userIds: ["lando", "ann"]) 
+controller.removeMembers(userIds: ["raya"])
+```
+
+:::caution
+You can only add/remove up to 100 members at once.
+:::
+
+### Leaving a channel
+
+It is possible for a user to leave the channel without moderator-level permissions if channel members have `RemoveOwnChannelMembership` permission:
```

Comment:
Please link the permissions page here

Reply:
Are you referring to this permission's page? 
https://getstream.io/chat/docs/ios-swift/chat_permission_policies/?language=swift&q=permissions

Reply:
Better to link https://getstream.io/chat/docs/ios-swift/channel_permission_policies/?language=swift&q=permissions

Or don't link at all and remove this sentence, since by default Channel Members already have this permission, this sentence creates confusion if the reader doesn't know about permissions. If they already configured special permissions, they probably know what they're doing anyway

Reply:
Makes sense!